stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/options/MessageOptionsOverlayDialogFragment.kt

```diff
@@ -140,22 +231,12 @@ internal class MessageOptionsOverlayDialogFragment : DialogFragment() {
         fun onReactionClick(message: Message, reactionType: String)
     }
 
-    companion object {
-        const val TAG = "messageOptions"
-
-        private const val ARG_MESSAGE = "message"
-        private const val ARG_OPTIONS_CONFIG = "optionsConfig"
-
-        fun newInstance(
-            messageItem: MessageListItem.MessageItem,
-            configuration: MessageOptionsView.Configuration
-        ): MessageOptionsOverlayDialogFragment {
-            return MessageOptionsOverlayDialogFragment().apply {
-                arguments = bundleOf(
-                    ARG_MESSAGE to MessageItemWrapper(messageItem),
-                    ARG_OPTIONS_CONFIG to configuration
-                )
-            }
-        }
-    }
+    internal class Handlers(
+        val threadReplyHandler: (Message) -> Unit,
+        val editClickHandler: (Message) -> Unit,
+        val flagClickHandler: (Message) -> Unit,
+        val muteClickHandler: (User) -> Unit,
+        val blockClickHandler: (User) -> Unit,
+        val deleteClickHandler: (Message) -> Unit
+    ) : Serializable
```

Comment:
I never knew that Kotlin function types are serlializable, this is kind of shocking.

This seems extremely unsafe, given that these are instances of classes that implement the function interface. The code in there can have all sorts of references to other external objects (Views, ViewModels, Fragments, Activities, etc.), and those can not possibly be retained through serialization (or think about process death!).

I'd prefer to use the method we have in `ReactionsOverlayDialogFragment` for setting listeners on a nested Fragment, by calling a setter on it. That seems like it'll break after a process restoration too, but still seems a lot safer.

Reply:
Let me take a look in the `ReactionsOverlayDialogFragment`

Reply:
Moved the handlers as a setter instead of serialising it.